melosys-eessi-app/src/main/java/no/nav/melosys/eessi/service/mottak/SedMottakService.java

```diff
@@ -29,25 +29,25 @@
 
 
     public void behandleSed(SedMottattHendelse sedMottattHendelse) {
-        sedMottattHendelseRepository.save(sedMottattHendelse);
+        var lagretHendelse = sedMottattHendelseRepository.save(sedMottattHendelse);
 
-        var sed = euxService.hentSed(sedMottattHendelse.getSedHendelse().getRinaSakId(),
+        final var sed = euxService.hentSed(lagretHendelse.getSedHendelse().getRinaSakId(),
                 sedMottattHendelse.getSedHendelse().getRinaDokumentId());
 
-
         try {
-            opprettJournalpost(sedMottattHendelse);
+            opprettJournalpost(lagretHendelse);
         } catch (SedAlleredeJournalførtException e) {
             log.info("Inngående SED {} allerede journalført", e.getSedID());
+            sedMottattHendelseRepository.delete(lagretHendelse);
             return;
         }
 
-        sedMottattHendelseRepository.save(sedMottattHendelse);
+        sedMottattHendelseRepository.save(lagretHendelse);
```

Comment:
Bare ved første blikk ser det ut som `lagretHendelse` lagres 2 ganger uten grunn. Det ville hjelpe forståelse og vedlikehold hvis hendelsene var immutable. For nå kunne man kanskje få `opprettJournalpost` til å returnere en jpID og flytte `sedMottattHendelse.setJournalpostId` for å synliggjøre endringen.

Reply:
Enig!